v2/component/launcher.go

```diff
@@ -354,29 +354,24 @@ func (l *Launcher) storeOutputArtifactMetadataFromCache(ctx context.Context, exe
 			continue
 		}
 		runtimeArtifact := runTimeArtifactList.Artifacts[0]
-		mlmdArtifacts, ok := mlmdOutputArtifactsByName[runtimeArtifact.GetName()]
-		if !ok || len(mlmdArtifacts) == 0 {
+		outputArtifact, ok := outputArtifacts[runtimeArtifact.GetName()]
+		if !ok || outputArtifact == nil {
```

Comment:
Is it possible that outputArtifact can be nil when ok is set to true?

Reply:
fair enough, agree output artifact != nil should be implied by ok=true